packages/react/src/FieldSet/FieldSet.tsx

```diff
@@ -6,25 +6,43 @@
 import clsx from 'clsx'
 import { forwardRef } from 'react'
 import type { ForwardedRef, HTMLAttributes, PropsWithChildren } from 'react'
+import { Hint, HintProps } from '../Hint'
 
 export type FieldSetProps = PropsWithChildren<HTMLAttributes<HTMLFieldSetElement>> & {
   /** Whether the field set has an input with a validation error */
   invalid?: boolean
   /** The text for the caption. */
   legend: string
-}
+  /** Whether the associated inputs are optional. Will append the text '(niet verplicht)' to the label if no hint is provided. */
+  optional?: boolean
+} & HintProps
 
 export const FieldSet = forwardRef(
-  ({ children, className, invalid, legend, ...restProps }: FieldSetProps, ref: ForwardedRef<HTMLFieldSetElement>) => (
-    <fieldset
-      {...restProps}
-      ref={ref}
-      className={clsx('ams-field-set', invalid && 'ams-field-set--invalid', className)}
-    >
-      <legend className="ams-field-set__legend">{legend}</legend>
-      {children}
-    </fieldset>
-  ),
+  (
+    { children, className, invalid, legend, hint, optional, ...restProps }: FieldSetProps,
+    ref: ForwardedRef<HTMLFieldSetElement>,
+  ) => {
+    let optionalHint = null
+
+    if (optional) {
+      optionalHint = hint ? <Hint hint={hint} /> : <Hint hint="niet verplicht" />
+    } else if (hint) {
+      optionalHint = <Hint hint={hint} />
+    }
```

Comment:
Can't we extract this logic to the Hint component itself? Something like this (pseudocode):

```
const getHintText = ({ hint, optional }: HintProps) => {
  if (hint) {
    return <Hint hint={hint} />
  } else if (optional) {
    return <Hint hint="niet verplicht" />
  }

  return null
}

export const Hint = forwardRef(
  ({ className, hint, optional, ...restProps }: HintProps, ref: ForwardedRef<HTMLElement>) => {
    const hintText = getHintText({ hint: hint, optional: optional })

    return (
      hintText && (
        <span {...restProps} ref={ref} className={clsx('ams-hint', className)}>
          ({hintText})
        </span>
      )
    )
  },
)
```